packages/react-native-bpk-theming/src/create-primary-color-extractor.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* @flow */
+import React, { type ComponentType } from 'react';
+
+import { withTheme } from './BpkThemeProvider';
+
+// TODO: Consider creating a single utility for this as it's used in many places
+const getDisplayName = Component =>
+  Component.displayName || Component.name || 'Component';
+
+export default (defaultValue: string) => <
```

Comment:
Not sure I understand the advantage of this approach instead of just a function to read the property. I'm not a big fan of HOCs, specially because here we can have `withPrimaryColor` and `withTheme` together, that's a bit unecessary IMO. 

If we have `withTheme` already I think we should just use that and provide a simple function that reads from it. I believe the fact that users will have to wrap the component with `withTheme` is not a big deal as the function will require the theme, and that would make it clear wrapping is required.

Another point is that this approach is not very flexible, adding a `secondaryColor` would be a lot more complex.

Reply:
I agree on your point about `secondaryColor` the reason we are going for this over just a function where the consumer can pass the `theme` prop from `withTheme` and have the function extract the primary color. The HOC is preferable over this method because:

1. Consumer has to do less work hence there's less risk to do it incorrectly.
2. We maintain abstraction and can change how we do theming without consumers having to care since they are not directly using `withTheme`